benchmarks/benchmark_dht.py

```diff
@@ -1,129 +1,164 @@
+
+        total_stores += len(store_result)
+        successful_stores += sum(map(bool, [
+            result.result() for result in store_result
+        ]))
+        await asyncio.sleep(wait_before_read)
+        
+        get_start = time.perf_counter()
+        get_peers = random.sample(peers, min(num_get_peers, len(peers)))
+        get_tasks = [
+            get_task(peer, key) for peer in get_peers
+        ]
+        get_result, _ = await asyncio.wait(
+            get_tasks, return_when=asyncio.ALL_COMPLETED
+        )
+        for result in get_result:
+            if result.result()[0]:
+                successful_gets += 1
+        
+        total_get_time += time.perf_counter() - get_start
+
+        await asyncio.sleep(wait_after_iteration)
+
+    logger.info(
+        f"Store success rate: {successful_stores / total_stores * 100:.1f}% \
+        ({successful_stores} / {total_stores})"
+    )
+    logger.info(f"Mean store time: {total_store_time / total_stores:.5}, \
+        Total: {total_store_time:.5}")
+    logger.info(
+        f"Get success rate: \
+        {successful_gets / num_get_peers / min(num_get_peers, len(peers)) * 100:.1f} \
+        ({successful_gets} / {num_get_peers * min(num_get_peers, len(peers))})"
+    )
+    logger.info(
+        f"Mean get time: \
+        {total_get_time / num_get_peers / total_num_rounds:.5f}, \
+        Total: {total_get_time:.5f}"
+    )
+            
 
 def benchmark_dht(
     num_peers: int,
     initial_peers: int,
-    num_experts: int,
-    expert_batch_size: int,
     random_seed: int,
-    wait_after_request: float,
+    num_threads: int,
+    total_num_rounds: int,
+    num_store_peers: int,
+    num_get_peers: int,
+    wait_after_iteration: float,
     wait_before_read: float,
     wait_timeout: float,
     expiration: float,
 ):
     random.seed(random_seed)
-
+    
     logger.info("Creating peers...")
     peers = []
     for _ in trange(num_peers):
         neighbors = sum(
-            [peer.get_visible_maddrs() for peer in random.sample(peers, min(initial_peers, len(peers)))], []
+            [peer.get_visible_maddrs() for peer in random.sample(
+                peers, min(initial_peers, len(peers))
+            )], []
         )
-        peer = hivemind.DHT(initial_peers=neighbors, start=True, wait_timeout=wait_timeout)
+        peer = hivemind.DHT(initial_peers=neighbors, start=True, 
+                wait_timeout=wait_timeout)
         peers.append(peer)
 
-    store_peer, get_peer = peers[-2:]
+    logger.info("Creating coroutines...")
+    loop = asyncio.get_event_loop()
 
-    expert_uids = list(
-        set(
-            f"expert.{random.randint(0, 999)}.{random.randint(0, 999)}.{random.randint(0, 999)}"
-            for _ in range(num_experts)
-        )
-    )
-    logger.info(f"Sampled {len(expert_uids)} unique ids (after deduplication)")
-    random.shuffle(expert_uids)
+    task_list = [
+        loop.create_task(
```

Comment:
this is not recommended, better make the whole function asynchronous and use `asyncio.run`